robusta_krr/core/integrations/kubernetes/__init__.py

```diff
@@ -47,6 +48,43 @@ def __init__(self, cluster: Optional[str]=None):
 
         self.__jobs_for_cronjobs: dict[str, list[V1Job]] = {}
         self.__jobs_loading_locks: defaultdict[str, asyncio.Lock] = defaultdict(asyncio.Lock)
+        self.__namespaces: Union[list[str, None]] = None
+
+    @property
+    def namespaces(self) -> Union[list[str], Literal["*"]]:
+        """wrapper for settings.namespaces, which will do expand namespace list if some regex pattern included
+
+        Returns:
+            A list of namespace that will be scanned
+        """
+        # just return the list if it's already initialized
+        if self.__namespaces != None:
+            return self.__namespaces
+
+        setting_ns = settings.namespaces
+        if setting_ns == "*":
+            self.__namespaces = setting_ns
+            return self.__namespaces
+
+        self.__namespaces = []
+        expand_list: list[re.Pattern] = []
+        ns_regex_chars = re.compile(r"[\\*|\(.*?\)|\[.*?\]|\^|\$]")
```

Comment:
Is there a reason to check for something regex-like first as opposed to always evaluating the namespace strings as regex?

Reply:
actually, this is for backward compatibility. since it requires additional grants to listing namespace as in previous [discussion](https://github.com/robusta-dev/krr/pull/336#discussion_r1770618925) . that's the reason i put the "notes" in the README.md